src/core/solution.jl

```diff
@@ -1,431 +1,445 @@
 import LinearAlgebra: tr
```

Comment:
There are tons of lines commented out in this file. Do you foresee any of it being necessary down the line? If not, go ahead and delete it.